### DISCUSSION TITLE
A-Frame support for easier WebAR/VR usage

### DISCUSSION THREAD

**Discussion by FireDragonGameStudio** [needle-tools/needle-engine-support#85] [Feature Ideas]
As long as there is no support for WebXR from Apple (Safari and Chrome on iOS), a workaround for AR experiences is needed. An idea would be to integrate A-Frame, as it's a layer built on three.js with wide support for WebAR solutions like AR.js and MindAR.

**Comment by hybridherbst** [comment]
Would you mind writing up a few workflows / systems that would get easier with A-Frame support so we understand this better? At least in my mind A-Frame is somewhere between vanilla three and react-three-fiber (also not sure about how active the devs are: last release in feburary, no blog post since 2020), but seems there's still more to it.

**Reply by FireDragonGameStudio** [reply]
Sure, when looking at the A-Frame github repo (https://github.com/aframevr/aframe/) it's still actively maintained, although the website doesn't reflect this ^^ When looking at WebAR (AR.js and MindAR), the A-Frame support is pretty good and easy to do (most of the A-Frame extensions use A-Frames three.js instead of shipping their own integration. These custom integrations makes it sometimes hard to use, when having a three.js only approach, to replace the integration with Needle references (for example with MindAR).

A possible workflow (at least in my imagination) would be to have A-Frame active only for the AR/VR stuff (not WebXR, although that would maybe make sense too). Like the WebXR parts in Unity, there could be an Unity A-Frame component, which rewires everything to the A-Frame components under the hood. Afaik it's currently smth like (very simplified) - Unity Editor -> Needle -> three.js. By introducing another layer like Unity -> Needle -> A-Frame, the three.js part could be done via A-Frame. 

I know it's not that simple, but maybe this'll make things easier, when integrating existing WebAR/VR solutions (e.g. Hand-Tracking)
